### ISSUE TITLE
Proposal: rely on duck typing to determine if input is a Jimp instance instead of comparing constructors

### ISSUE THREAD

**Issue by Iwasawafag** [oliver-moran/jimp#200]
I think it would be great if we were able to instantiate Jimp image from another jimp-like object whose constructor is not necessary the same as constructor from the current instance.

And personally I think of one remarkable feature - `bitmap` object. It is a part of public API (lowlevel, but still public), was around for ever and unlikely to be changed in the foreseen future. It also contains all necessary data to make a copy of an image. So I would suggest adding a method that looks something like this

```js
Jimp.isJimp = function(image) {
    if (image && image.constructor === Jimp) return true;

    return image && image.bitmap 
        && typeof image.bitmap.width === 'number'
        && typeof image.bitmap.height === 'number'
        && Buffer.isBuffer(image.bitmap.data) 
        && image.bitmap.data.length === image.bitmap.width * image.bitmap.height * 4;
}
```

and using it in the constructor instead of current check `if ("object" == typeof arguments[0] && arguments[0].constructor == Jimp) {`

## Why?

So different modules could pass instances of Jimp around. Because otherwise we would have to encode image to something like PNG in one module just to decode it right away in another. Encoding/decoding takes time and wastes CPU cycles we don't want to waste.

It two or more modules in the project share same dependency – it would be rational thing to open/create and preprocess image in one module and send it as is to another in RAW format, if that other module known to accept Jimp instances as image sources. But the highter number of such packages used at the same time (think of something with modular architecture or a Gulp/Grunt-like toolchain) — the lower chances are that all of them will depend on the same version of Jimp.

**Comment by Iwasawafag** [comment]
Or rather it should just accept raw pixel data object as first argument to the constructor. It's a commonly used across different libraries and will make hooking alternative/new decoders little easier

```js
var Jimp = require('jimp'),
    jpeg = require('jpeg-js'),
    fs = require('fs');

var jpegData = fs.readFileSync('grumpycat.jpg');
var rawImageData = jpeg.decode(jpegData, true);
var jimpImage1 = new Jimp(rawImageData);

// and then

var image2 = new Jimp(jimpImage1.bitmap);
```

**Comment by hipstersmoothie** [comment]
closed via #514
